tester/src/test/scala/spinal/tester/scalatest/FormalArbiterTester.scala

```diff
@@ -0,0 +1,455 @@
+        val output = master(Stream(dataType))
+
+        select := dut.io.chosen
+        selectOH := dut.io.chosenOH
+        output << dut.io.output
+
+        when(reset || past(reset)) {
+          for (i <- 0 until portCount) {
+            assume(inputs(i).valid === False)
+          }
+          assert(!dut.locked)
+          assert(select === 0)
+        }
+
+        for (i <- 0 until portCount) {
+          inputs(i) >> dut.io.inputs(i)
+          inputs(i).formalAssumesSlave()
+        }
+        output.formalCovers(2)
+        output.formalAssertsMaster()
+
+        assert(select < portCount)
+        assert(select === OHToUInt(selectOH))
+        assert(output.payload === inputs(select).payload)
+
+        // if not locked or fragment last fired, select will be lowest inputs.valid
+        val allInputsValidOH = OHMasking.first(Vec(inputs.map(_.valid))).asBits
+        val selectRefresh = !dut.locked || past(output.last && output.fire)
+        cover(selectRefresh)
+        when(selectRefresh) {
+          assert(selectOH === allInputsValidOH)
+        }
+
+        // chosenOH should be a OneHot value
+        assert(selectOH === OHMasking.first(selectOH))
+
+        val shouldStableSelCond = !past((allInputsValidOH === 0) || output.last || reset)
+        cover(shouldStableSelCond)
+        when(shouldStableSelCond) {
+          assert(stable(select))
+        }
+      })
+  }
+
+  test("Arbiter-roundrobin-none-verify") {
+    FormalConfig
+      .withBMC(20)
+      .withProve(20)
+      .withCover(20)
+      // .withDebug
+      .doVerify(new Component {
+        val portCount = 5
+        val dataType = Bits(8 bits)
+        val dut = FormalDut(
+          new StreamArbiter(dataType, portCount)(StreamArbiter.Arbitration.roundRobin, StreamArbiter.Lock.none)
+        )
+        val reset = ClockDomain.current.isResetActive
+
+        assumeInitial(reset)
+
+        val select = UInt(log2Up(portCount) bit)
+        val selectOH = Bits(portCount bit)
+        val inputs = Vec(slave(Stream(dataType)), portCount)
+        val output = master(Stream(dataType))
+
+        select := dut.io.chosen
+        selectOH := dut.io.chosenOH
+        output << dut.io.output
+
+        when(reset || past(reset)) {
+          for (i <- 0 until portCount) {
+            assume(inputs(i).valid === False)
+          }
+        }
+
+        for (i <- 0 until portCount) {
+          inputs(i) >> dut.io.inputs(i)
+          inputs(i).formalAssumesSlave()
+        }
+        output.formalCovers(2)
+
+        assert(select < portCount)
+        assert(select === OHToUInt(selectOH))
+        assert(output.payload === inputs(select).payload)
+        assert(output.fire === inputs(select).fire)
+
+        val requests = Vec(inputs.map(_.valid)).asBits.asUInt
+        val uGranted = dut.io.chosenOH.asUInt
+        val doubleRequests = requests @@ requests
+        val doubleGrant = doubleRequests & ~(doubleRequests - uGranted)
```

Comment:
Is this to grant the lowFirst port of the rest of the ports other than the selected port?